src/canvasObject.js

```diff
@@ -1,159 +1,128 @@
 'use strict';
 
 require('openseadragon');
+var ImageResource = require('./ImageResource');
 
 var CanvasObject = function(config, dispatcher) {
-  this.fullyOpened = config.fullyOpened || false;
-  this.visible = config.visible || true; // todo: this is not used yet. Do we need it?
+  var self = this;
   this.clipRegion = config.clipRegion;
   this.opacity = config.opacity || 1;
-  this.x = config.x || 0;
-  this.y = config.y || 0;
-  this.placeholder = config.placeholder || { type: 'image', url: './example-thumbnail.png' };
   this.index = config.index;
 
   this.id = config.canvas['@id'];
-  this.height = config.canvas.height;
-  this.width = config.canvas.width;
-  this.images = config.canvas.images;
+
+  this.bounds = {
+    x : config.x || 0,
+    y : config.y || 0,
+    height : config.canvas.height,
+    width : config.canvas.width
+  };
+  this.thumbUrl = config.canvas.thumbnail;
+  this.thumbService = config.canvas.images[0].resource.service['@id'];
```

Comment:
As we've seen, not every canvas has a `service`. I suppose this is part of what @aeschylus wants us to do with the factory he's proposing, but maybe we shouldn't hold off until that is all spec'd out. 

It looks like in the manifests that don't have service, there is an image located at: 

config.canvas.images[0].resource['@id']

...so you could grab that instead and use it to make an ImageTileSource rather than an IIIFTileSource inside the ImageResource. 

At the very least it might be good to check whether the thing you're looking for exists, and if it doesn't, fail gracefully rather than throwing an exception (which is what it does now). 

Same goes for `this.images` below.